crates/agent/src/api/handlers.rs

```diff
@@ -6,10 +6,24 @@ use axum::{
     Json,
 };
 use serde::Serialize;
+use utoipa::ToSchema;
 
 use crate::{api::routes::AppState, models::Proposal, services::agent::AgentServiceTrait};
 
 /// Analyze a proposal
+#[utoipa::path(
+    post,
+    path = "/analyze",
+    request_body = Proposal,
+    responses(
+        (status = 200, description = "Analysis completed successfully", body = AnalyzeResponse),
+        (status = 400, description = "Invalid request data"),
+        (status = 500, description = "Internal server error during analysis")
+    ),
+    tag = "Analysis",
+    summary = "Analyze a DAO/Governance proposal",
+    description = "Submit a proposal for AI analysis. The service will evaluate the proposal content and provide insights on governance quality, risks, and recommendations."
+)]
```

Comment:
Is there a way to move these response examples into a separate file? The current setup looks fine, but if we plan to add multiple examples in the future, keeping them inline could reduce the overall code quality.